packages/wordcount/src/index.js

```diff
@@ -50,56 +61,56 @@ function loadSettings( type, userSettings ) {
 }
 
 /**
- * Match the regex for the type 'words'
+ * Count the words in text
  *
- * @param {string} text     The text being processed
- * @param {string} regex    The regular expression pattern being matched
- * @param {Object} settings Settings object containing regular expressions for each strip function
+ * @param {string}              text     The text being processed
+ * @param {RegExp}              regex    The regular expression pattern being matched
+ * @param {WPWordCountSettings} settings Settings object containing regular expressions for each strip function
  *
- * @return {Array|{index: number, input: string}} The matched string.
+ * @return {number} The matched string.
```

Comment:
```suggestion
 * @return {number} The number of words in the text.
```

(or `matches` instead of `words`).